### PR TITLE
Enable shellcheck [ECR-4153]

### PR DESCRIPTION
**Pull request by dmitry-timofeev**
## Overview

Enable shellcheck and fix shellcheck issues.

You can see the fixed issues by running

```shell
brew install shellcheck
git checkout master
find . -iname '*.sh' -exec shellcheck {} + 
```

---

### Definition of Done

- [ ] There are no TODOs left in the code
- [ ] Change is covered by automated [tests](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#tests)
- [ ] The [coding guidelines](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [ ] Public API has Javadoc
- [ ] Method preconditions are checked and documented in the Javadoc of the method
- [ ] Changelog is updated if needed (in case of notable or breaking changes)
- [ ] The continuous integration build passes

### CODE REVIEW

**Comment by dmitry-timofeev** [comment]
It actually uncovered some broken scripts:

This used to work *even when the shell had error flag set* (see [Travis output from master](https://travis-ci.com/exonum/exonum-java-binding/jobs/290863721#L384)):

https://github.com/exonum/exonum-java-binding/blob/7b6d2603cc5e1973a2fa74be62517daf6f3387e6/exonum-java-binding/tests_profile#L56

The shellcheck complained about that: https://github.com/koalaman/shellcheck/wiki/SC2155

And now we have a failing build: https://travis-ci.com/exonum/exonum-java-binding/jobs/291299139#L396

File: .shellcheckrc [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Allow which usage
```
Comment (bullet-tooth) [inline .shellcheckrc]: I would leave a link to the guide here. To have a quick reference and find rules, for example, `SC2230` :)

Comment (dmitry-timofeev) [reply .shellcheckrc]: Good, added

File: .shellcheckrc [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# See https://github.com/koalaman/shellcheck/wiki/Checks
+
+# Allow 'which' usage
+disable=SC2230
+
+# Suppress "Not following: ./tests_profile: openBinaryFile: does not exist (No such file or directory)"
+disable=SC1091
```
Comment (bullet-tooth) [inline .shellcheckrc]: minor: EOF

**Pull request merged**